### PR TITLE
feat: Added and improved errors and warnings in the table transformers

### LINKED ISSUE #152: Improve error handling of table transformers
### Is your feature request related to a problem?

In some cases, error handling is not ideal (see solution for details) in the `safeds.data.tabular.transformation` package.

### Desired solution

* Validate inputs of a function as the first step
* Raise an appropriate exception with an appropriate message if issues are found
* Document in the docstring that the exception is raised and under which condition
* Test that the correct exception with the correct message is raised

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by Marsmaennchen221**
Closes #152.

### Summary of Changes

feat: Added and improved errors and warnings in the table transformers
feat: Added ability for the `Imputer` with strategy `Mode` to work if multiple most frequent values (warns if this is the case, takes the lowest most frequent value)
feat: Added ability for the `OneHotEncoder` to encode python NaNs
fix: fixed grammar in `NonNumericColumnError`
feat: Changed `ValueNotPresentWhenFittedError` to allow info about multiple columns
test: Changed test for `Imputer` to test all strategies

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/7.0.4) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/5304704065)

| Descriptor  |                                Linter                                 |Files|Fixed|                            Errors                             |Elapsed time|
|-------------|-----------------------------------------------------------------------|-----|----:|---------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/7.0.4/descriptors/python_black)          |   14|   11|                                                              0|2.75s       |
|❌ PYTHON    |[mypy](https://megalinter.io/7.0.4/descriptors/python_mypy)            |   14|     |[1](https://github.com/Safe-DS/Stdlib/actions/runs/5304704065) |2.66s       |
|❌ PYTHON    |[ruff](https://megalinter.io/7.0.4/descriptors/python_ruff)            |   14|   11|[73](https://github.com/Safe-DS/Stdlib/actions/runs/5304704065)|0.05s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/7.0.4/descriptors/repository_git_diff)|yes  |     |no                                                             |0.01s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/5304704065)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

**Review by patrikguempel — APPROVED** [review]

**Review by jxnior01 — APPROVED** [review]

**Review by sibre28 — APPROVED** [review]
[LGTM](https://lgtm.party/p/EA5K3PG/)

File: src/safeds/data/tabular/transformation/_imputer.py [hunk 1]
```diff
@@ -107,18 +108,59 @@ def fit(self, table: Table, column_names: list[str] | None) -> Imputer:
...
+                    sorted(
+                        set(table.keep_only_columns(column_names).column_names)
+                        - set(
+                            table.keep_only_columns(column_names)
+                            .remove_columns_with_non_numerical_values()
+                            .column_names,
+                        ),
+                    ),
+                ),
+            )
```
Comment (robmeth) [inline src/safeds/data/tabular/transformation/_imputer.py:147]: maybe put the NonNumericColumnError before the ValueError to keep the order of Errors the same as in the other transformations

**Pull request merged**